src/GLDraw.ts

```diff
@@ -25,56 +30,19 @@ export enum CAP {
   */
 export namespace GLDraw {
 
-    // Rotation matrix around z and x axis -
-    // according to y basis vector
-    // TODO: Try to optimize this (square roots?)
+    // Rotation matrix around z and x axis - according to y basis vector
     function getRotationMatrix(dx: number, dy: number, dz: number) {
-        var dxy = Math.sqrt(dx * dx + dy * dy);
-        var dyz;
-
-        var sinA, cosA, sinB, cosB;
-
-        // about z axis - Phi
-        if (dxy < 0.0001) {
-            sinA = 0;
-            cosA = 1;
-        }
-
-        else {
-            sinA = -dx / dxy;
-            cosA = dy / dxy;
-        }
-
-        // recast dy in terms of new axes - z is the same
+        // Using Math.hypot(dx,dy) instead of Math.sqrt(dx * dx + dy * dy)
+        const dxy = Math.hypot(dx, dy);
+        const { sin: sinA, cos: cosA } = dxy < 0.0001 ? { sin: 0, cos: 1 } : { sin: -dx / dxy, cos: dy / dxy };
 
         dy = -sinA * dx + cosA * dy;
-        dyz = Math.sqrt(dy * dy + dz * dz);
-
-        // about new x axis - Theta
+        const dyz = Math.hypot(dy, dz);
+        const { sin: sinB, cos: cosB } = dyz < 0.0001 ? { sin: 0, cos: 1 } : { sin: dz / dyz, cos: dy / dyz };
 
-        if (dyz < 0.0001) {
-            sinB = 0;
-            cosB = 1;
-        }
-
-        else {
-            sinB = dz / dyz;
-            cosB = dy / dyz;
-        }
-
-        var rot = new Float32Array(9);
-        rot[0] = cosA;
-        rot[1] = sinA;
-        rot[2] = 0;
-        rot[3] = -sinA * cosB;
-        rot[4] = cosA * cosB;
-        rot[5] = sinB;
-        rot[6] = sinA * sinB;
-        rot[7] = -cosA * sinB;
-        rot[8] = cosB;
+        const rot = new Float32Array([cosA, sinA, 0, -sinA * cosB, cosA * cosB, sinB, sinA * sinB, -cosA * sinB, cosB]);
```

Comment:
Setting the values individually instead of allocating a list first is more than twice as fast (go ahead and run it through JSBench.Me)